### PR TITLE
Migrate Figure.coast tests to dvc

### PR DESCRIPTION
**Pull request by willschlitzer**
Changing the tests in test_coast.py to use `dvc` and `mpl_image_compare`. An update/fix to #1102.

Related to #1131.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_coast.py [hunk 1]
```diff
@@ -14,60 +13,37 @@ def test_coast():
...
-        B=5,
-        I="1/1p,blue",
-        N="1/0.25p,-",
-        W="0.25p,white",
-        G="green",
-        S="blue",
-        D="c",
-        A=10000,
+        region="-30/30/-40/40",
+        projection="m0.1i",
```
Comment (weiji14) [inline pygmt/tests/test_coast.py:17]: Good opportunity to change to SI units, and use a Python list for `region`.
```suggestion
        region=[-30, 30, -40, 40],
        projection="m0.25c",
```

File: pygmt/tests/test_coast.py [hunk 2]
```diff
@@ -14,60 +13,37 @@ def test_coast():
...
-    )
-    return fig_ref, fig_test
+    fig = Figure()
+    fig.coast(region=[-30, -10, 60, 65], projection="m1c", frame=True, land="p28+r100")
+    return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_coast_aliases():
```
Comment (weiji14) [inline pygmt/tests/test_coast.py:41]: This is one of the top 3 slowest tests as reported in #840, can you try and speed it up? E.g. using resolution="c" for crude? Maybe also plot a smaller geographical region.

Comment (willschlitzer) [reply pygmt/tests/test_coast.py:41]: Done 2f7ba86 and 0fcbdfc

**Comment by seisman** [comment]
The [`coast` wrapper](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/src/coast.py) is very simple and only has a few lines of codes. 

As documented in the [contributing guides](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#testing-your-code),

> The highest priority for tests should be the Python-specific code, such as numpy, pandas, and xarray objects and the virtualfile mechanism.

I feel some of the tests are unnecessary. Tests for a global map, a regional map, a map using `dcw` should be enough.

**Comment by willschlitzer** [comment]
> The [`coast` wrapper](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/src/coast.py) is very simple and only has a few lines of codes. 
> 
> 
> 
> As documented in the [contributing guides](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#testing-your-code),
> 
> 
> 
> > The highest priority for tests should be the Python-specific code, such as numpy, pandas, and xarray objects and the virtualfile mechanism.
> 
> 
> 
> I feel some of the tests are unnecessary. Tests for a global map, a regional map, a map using `dcw` should be enough.
> 
> 
> 
> Edit: IMHO, only the following tests are needed:
> 
> 
> 
> - test_coast_iceland
> 
> - test_coast_world_mercator
> 
> - test_coast_required_args
> 
> - test_coast_dcw_single
> 
> - test_coast_dcw_list

Sounds good! I'll shorten it down.

**Comment by willschlitzer** [comment]
@seisman @weiji14 I removed the tests that we deemed not needed and deleted the `.dvc` files; do I have to do anything else to fully remove these tests?

File: pygmt/tests/test_coast.py [hunk 3]
```diff
@@ -1,94 +1,21 @@
...
 from pygmt.exceptions import GMTInvalidInput
-from pygmt.helpers.testing import check_figures_equal
 
 
 @pytest.mark.mpl_image_compare
-def test_coast():
+def test_coast_iceland():
     """
-    Simple plot from the GMT docs.
+    Test passing in region as a list.
```
Comment (seisman) [inline pygmt/tests/test_coast.py:12]:
```suggestion
    Test plotting a regional map with coastlines.
```

File: pygmt/tests/test_coast.py [hunk 4]
```diff
@@ -1,94 +1,21 @@
...
-        transparency=13,  # t
-        dcw="MA+gred",  # E
-        lakes="blue",  # C
-    )
-    return fig_ref, fig_test
-
-
 @pytest.mark.mpl_image_compare
 def test_coast_world_mercator():
     """
```
Comment (seisman) [inline pygmt/tests/test_coast.py:21]: > Test passing generating a global Mercator map with coastlines.

should be changed to 

> Test generating a global Mercator map with coastlines.

File: pygmt/tests/test_coast.py [hunk 5]
```diff
@@ -1,94 +1,21 @@
...
-        perspective=[135, 25],  # p
-        transparency=13,  # t
-        dcw="MA+gred",  # E
-        lakes="blue",  # C
-    )
-    return fig_ref, fig_test
-
-
 @pytest.mark.mpl_image_compare
 def test_coast_world_mercator():
```
Comment (seisman) [inline pygmt/tests/test_coast.py:20]: Please change `projection="M10i",` to "projection="M15c",".

Also change `resolution="l",` to `resolution="c",`.

Both should help decrease the image size.

File: pygmt/tests/test_coast.py [hunk 6]
```diff
@@ -1,94 +1,21 @@
...
 """
 import pytest
 from pygmt import Figure
 from pygmt.exceptions import GMTInvalidInput
-from pygmt.helpers.testing import check_figures_equal
 
 
 @pytest.mark.mpl_image_compare
-def test_coast():
+def test_coast_iceland():
```
Comment (seisman) [inline pygmt/tests/test_coast.py:10]: I think we should rewrite this test. Below is the baseline image for this test. It's difficult to verify its correctness. Instead, if we plot a regional map like Japan, Hawaii, or Australia, it's much easier to check.

image

File: pygmt/tests/test_coast.py (hunk 5 above)
Comment (willschlitzer) [reply pygmt/tests/test_coast.py:20]: Updated 9ad494e

File: pygmt/tests/test_coast.py (hunk 6 above)
Comment (willschlitzer) [reply pygmt/tests/test_coast.py]: In 150c45e, I changed the region (had to pick Japan; I miss Okinawa!) and think I made it more readable.

File: pygmt/tests/test_coast.py (hunk 4 above)
Comment (willschlitzer) [reply pygmt/tests/test_coast.py:21]: Updated

File: pygmt/tests/baseline/test_coast_iceland.png.dvc [hunk 7]
```diff
@@ -0,0 +1,4 @@
+outs:
```
Comment (seisman) [inline pygmt/tests/baseline/test_coast_iceland.png.dvc:1]: This file `pygmt/tests/baseline/test_coast_iceland.png.dvc` should be deletete.

Comment (seisman) [reply pygmt/tests/baseline/test_coast_iceland.png.dvc:1]: Ping @willschlitzer to address the comment and then this PR should be ready for merge.

Comment (willschlitzer) [reply pygmt/tests/baseline/test_coast_iceland.png.dvc]: Done in 0a1c5e7

**Review by seisman — APPROVED** [review]

**Pull request merged**